<!DOCTYPE html><html lang="en"><head><title>Static News</title><meta charSet="utf-8"/><meta name="description" content="Static delayed Hacker News."/><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"/><meta name="theme-color" media="(prefers-color-scheme: dark)" content="#1d1f21"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><meta name="application-name" content="Static News"/><meta name="apple-mobile-web-app-title" content="Static News"/><meta name="apple-mobile-web-app-capable" content="yes"/><meta name="mobile-web-app-capable" content="yes"/><meta name="apple-mobile-web-app-status-bar-style" content="#1d1f21"/><link rel="preload" href="styles.css?v=1728550878897" as="style"/><link rel="stylesheet" href="styles.css?v=1728550878897"/></head><body><div id="container"><div id="inner"><header><a href="/">Static News</a><a href="/about">about</a></header><div id="content"><div><div id="title"><a href="https://blog.miguelgrinberg.com/post/the-ultimate-guide-to-error-handling-in-python">The Ultimate Guide to Error Handling in Python</a>Â <span class="domain">(<a href="https://blog.miguelgrinberg.com">blog.miguelgrinberg.com</a>)</span></div><div class="subtext"><span>kdamica</span> | <span>4 comments</span></div><br/><div><div id="41796770" class="c"><input type="checkbox" id="c-41796770" checked=""/><div class="controls bullet"><span class="by">atoav</span><span>|</span><a href="#41796618">next</a><span>|</span><label class="collapse" for="c-41796770">[-]</label><label class="expand" for="c-41796770">[2 more]</label></div><br/><div class="children"><div class="content">You can do nearly everything using Python errors. One thing that bothers me a bit, is the amount of text needed to catch them and bubble them up. Depending on the programming style traditional exceptions may also not be adequately re<p>For this Rust style error handling can be interesting, eg using poltergeist (<a href="https:&#x2F;&#x2F;github.com&#x2F;alexandermalyga&#x2F;poltergeist">https:&#x2F;&#x2F;github.com&#x2F;alexandermalyga&#x2F;poltergeist</a>) or result (<a href="https:&#x2F;&#x2F;pypi.org&#x2F;project&#x2F;result&#x2F;" rel="nofollow">https:&#x2F;&#x2F;pypi.org&#x2F;project&#x2F;result&#x2F;</a>).</div><br/><div id="41796968" class="c"><input type="checkbox" id="c-41796968" checked=""/><div class="controls bullet"><span class="by">drcongo</span><span>|</span><a href="#41796770">parent</a><span>|</span><a href="#41796618">next</a><span>|</span><label class="collapse" for="c-41796968">[-]</label><label class="expand" for="c-41796968">[1 more]</label></div><br/><div class="children"><div class="content">Would you mind explaining what the benefits of either of those libraries is over normal exception handling? The examples at the top of the Result readme seem almost exactly as verbose as each other. I kinda like the idea of a Result object, but it seems like a lot of library imports for no gain so I&#x27;m sure I&#x27;m missing something.</div><br/></div></div></div></div><div id="41796618" class="c"><input type="checkbox" id="c-41796618" checked=""/><div class="controls bullet"><span class="by">InDubioProRubio</span><span>|</span><a href="#41796770">prev</a><span>|</span><label class="collapse" for="c-41796618">[-]</label><label class="expand" for="c-41796618">[1 more]</label></div><br/><div class="children"><div class="content">It still has exceptions- but exceptions are normal. Errors are normal.
How would one do : <a href="https:&#x2F;&#x2F;fsharpforfunandprofit.com&#x2F;rop&#x2F;" rel="nofollow">https:&#x2F;&#x2F;fsharpforfunandprofit.com&#x2F;rop&#x2F;</a> in python?</div><br/></div></div></div></div></div></div></div></body></html>